src/App.tsx

```diff
@@ -101,45 +129,275 @@ class App extends React.Component {
+
+  deleteMultipleMovie() {
+    console.log("delete movie")
+    this.setState({radioDisplay: true});
+  }
+
+
+  // menu item on cards
+  handleMenuClick = (event:any) => {
+    this.setState({ anchorEl: event.currentTarget });
+  };
+
+  checkBoxToggle = () => {
+    this.setState({checkBox: !this.state.checkBox})
+  }
 
+  render() { 
+    const { anchorEl } = this.state;
     return (
       <React.Fragment>
-      <AppBar position="sticky">
-        <Toolbar>
-          <CameraIcon />
-          <Typography variant="h6" color="inherit" noWrap>
-            Helium UI
-          </Typography>
-        </Toolbar>
-      </AppBar>
-      <main>
+      <div className="appbar">
+        <AppBar position="sticky">
+          <Toolbar>
+            <img src={Balloon} width="50" height="50" />
+            <Typography variant="h6" color="inherit" noWrap>
+              Helium UI
+            </Typography>          
+            <div className="searchBar">
+                <SearchIcon />
+                <InputBase
+                  placeholder="Search…"
+                />
+            </div>
+          </Toolbar>
+        </AppBar>          
+      </div>
+      <main> 
       <Grid container spacing={8}>           
           {this.state.movies.map((item, i) => (
             <Grid item key={i} sm={6} md={4} lg={3}>
-            <Card>
-              <CardContent>
-                  <Typography gutterBottom variant="h5" component="h2">
-                    {item.title}
-                  </Typography>
+              <Card className={item.title}>
+                <CardHeader 
+                  title = {item.title}
+                  action = {
+                    <IconButton 
+                      onClick={this.handleMenuClick}
+                      aria-owns={anchorEl ? 'cardMenu' : undefined}
+                      aria-haspopup="true"
+                    >
+                      <MoreVertIcon />
+                    </IconButton>
+                  }
+                />
+                <CardMedia
+                  style={{height: 0, paddingTop: '56.25%'}}
+                  image={MoviePH}
+                  title="img"
+                />
+                <CardContent>
                   <Typography>
                     Year: {item.year}<br />
                     Runtime: {item.runtime}min <br />
                     Genres: {this.joinStr(item.genres)}<br />
                   </Typography>
                 </CardContent>
+                <CardActions>
+                  <Checkbox 
+                    checked={this.state.checkBox}/>
+                </CardActions>
               </Card>
-              </Grid>
+            </Grid>
           ))}
-
         </Grid>
+        <Menu
+          id="cardMenu"
+          anchorEl={anchorEl}
+          open={Boolean(anchorEl)}
+          onClose={() => this.setState({ anchorEl: null })}  
+        >
+        <MenuItem onClick={() => this.setState({deleteDialog: true })}>Delete</MenuItem>
```

Comment:
do you think having these functions in-line might make them harder to find compared to having them as methods on the component object?

Reply:
I think is cleaner to have this way since all the method would be doing is changing and updating state. It saves a lot of lines of extra code that isn't necessarily needed. If there was more functionality, I would separate them. I almost think it is easier to find them this way since everything is included in the same place!